File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    

    for(let i=0; i<s.length; i++){
        if(s[i]==='('){
            stack.push(s[i]);
        }
        else{
            if(stack.length===0 || stack[stack.length-1]!=='(') return false;
            stack.pop();
        }
    }
    
    if(stack.length) return false;
    
    return true;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for i in range(len(s)):
        c = s[i]
        
        if c=='(':
            stack.append(c)
        else:
            if len(stack) == 0:
                return False
            last = stack.pop()
            if last == ')':
                return False
    
    if len(stack) != 0:
        return False

    return True